spec/inheritance_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'active_support'
+require 'active_record'
+require 'property_sets'
+
+ENV["RAILS_ENV"] = "test"
+
+yaml_config = "spec/support/database.yml"
+ActiveRecord::Base.configurations = begin
+                                      YAML.safe_load(IO.read(yaml_config), aliases: true)
+                                    rescue ArgumentError
+                                      YAML.safe_load(IO.read(yaml_config))
+                                    end
+
+class AbstractUnshardedModel < ActiveRecord::Base
+  self.abstract_class = true
+
+  connects_to database: {writing: :unsharded_database, reading: :unsharded_database_replica}
+end
+
+class Account < AbstractUnshardedModel
+  property_set :settings do
+    property :type
+  end
+end
```

Comment:
When defining models in the global scope, perhaps we should move them into spec/support/? Or name them something less likely to be defined elsewhere too? (We already have an Account model in specs, right?)